modules/metastorage-common/src/main/java/org/apache/ignite/metastorage/common/Cursor.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.metastorage.common;
+
+/**
+ * Closeable cursor.
+ *
+ * @param <T>
+ */
+//TODO: Should be replaced by common entity from org.ignite.lang or org.ignite.core package.
+public interface Cursor<T> extends Iterable<T>, AutoCloseable {
```

Comment:
Do you mean Cursor inherits Iterator?

For now, it is unclear what if one calls Cursor.iterator() twice? Will it same iterator or not?

Reply:
Will be revised under addition `ignite-core` module.